[ews-build] ValidateCommitMessage incorrectly fails when contributors.json is accessed from disk

briannafan opened the pull request:
#### 18eb1e59acc29712ce9e982dbab2af8a6a163bcc
<pre>
[ews-build] ValidateCommitMessage incorrectly fails when contributors.json is accessed from disk
<a href="https://bugs.webkit.org/show_bug.cgi?id=275519">https://bugs.webkit.org/show_bug.cgi?id=275519</a>
<a href="https://rdar.apple.com/129880497">rdar://129880497</a>

Reviewed by Aakash Jain.

Adds logging when we try to access contributors.json from disk.
Adds a check to Contributors.load after loading from disk to remove errors upon success.

* Tools/CISupport/ews-build/steps.py:
(Contributors.load):
(RunWebKitTests.evaluateResult):
(ValidateCommitMessage.run):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/280634@main">https://commits.webkit.org/280634@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e95be5ba2363f85271ac938beb6bb8b9412880d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56383 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35709 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8855 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/59990 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6819 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58509 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43331 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7013 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45674 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4765 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58412 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33582 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48652 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26536 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30362 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5982 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5823 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52352 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6254 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61674 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/291 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6371 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52934 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/56075 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/291 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48718 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/52826 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/259 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8525 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31536 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32622 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33705 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32369 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -6564,28 +6565,25 @@ def run(self, BufferLogObserverClass=logobserver.BufferLogObserver):
             self.summary = log_text.split('\n')[0]  # Display the first error if there are multiple
             rc = FAILURE
-        elif rc == SUCCESS:
+        if 'contributors.json' in self.summary and self.contributors:
```
How do we get the text 'contributors.json' inside the summary, is it coming from a previous error message. If so, that doesn't seem like a reliable way. Can we modify the underlying code so that rc is indeed SUCCESS when we are able to load from the disk (instead of us manually setting rc to SUCCESS).

briannafan replied:
Yeah it comes from the error message. Since it's not a shell command, I set it so that `Contributors.load` will reliably return no errors if it succeeds and that will work with the existing logic with the logs.

briannafan committed:
[ews-build] ValidateCommitMessage incorrectly fails when contributors.json is accessed from disk
https://bugs.webkit.org/show_bug.cgi?id=275519
rdar://129880497

Reviewed by Aakash Jain.

Adds logging when we try to access contributors.json from disk.
Adds a check to Contributors.load after loading from disk to remove errors upon success.

* Tools/CISupport/ews-build/steps.py:
(Contributors.load):
(RunWebKitTests.evaluateResult):
(ValidateCommitMessage.run):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/280634@main
